src/garage/torch/algos/batch_polopt.py

```diff
@@ -0,0 +1,157 @@
+"""A batch-based algorithm interleaves sampling and policy optimization."""
+import abc
+import collections
+
+from dowel import tabular
+import numpy as np
+
+from garage import log_performance, TrajectoryBatch
+from garage.misc import tensor_utils
+from garage.np.algos.base import RLAlgorithm
+from garage.sampler import OnPolicyVectorizedSampler
+from garage.tf.samplers import BatchSampler
+
+
+class BatchPolopt(RLAlgorithm):
```

Comment:
This is because there is `self.baseline` attribute in the `garage.tf.algos.batch_polopt`.
I need to rename the attribute to `self.value_function`
or maybe I just simple not inherit `BatchPolop`.

Reply:
The easiest way to fix this for now would just be to create a property (either in VPG or the original `BatchPolopt`) that makes `baseline` and `_value_function` the same. Alternatively, you could not inherit from `BatchPolopt`, and split its methods off the class into functions.

Reply:
Fixed